### ISSUE TITLE
Setting the Seed.

### ISSUE THREAD

**Issue by shivam-chandhok** [facebookresearch/DomainBed#69]
Hi,
I am unable to clearly understand the seed setup.
If I understand correctly, you have set the various seeds once in the beginning in train.py and then multiple runs are executed using the same process. 
Is this correct or my interpretation is wrong.
Can you please elaborate on the exact protocol used.

**Comment by lopezpaz** [comment]
There are two main seeds:

- The **hyper-parameter seed** decides the hyper-parameter values (learning rate, weight decay) for the algorithm.
- The **trial seed** decides everything else (initial weights, data partitions).

When sweeping, for each hyper-parameter seed we run a few trial seeds. This allows us to perform model selection across hyper-parameter seeds taking into account standard deviation (arising from the trial seeds, given each hyper-parameter seed).

**Comment by bad-meets-joke** [comment]
> There are two main seeds:
> 
> * The **hyper-parameter seed** decides the hyper-parameter values (learning rate, weight decay) for the algorithm.
> * The **trial seed** decides everything else (initial weights, data partitions).
> 
> When sweeping, for each hyper-parameter seed we run a few trial seeds. This allows us to perform model selection across hyper-parameter seeds taking into account standard deviation (arising from the trial seeds, given each hyper-parameter seed).

Hi, I do not think so according to the code.
https://github.com/facebookresearch/DomainBed/blob/93ebb6bccdb46d40471ce0713a165aaf7b5e21b6/domainbed/scripts/train.py#L75-L79
From line 83-84, the hparams are controlled by both **hyper-parameter seed** and **trial seed**, which means that for each hyper-parameter seed we **do not** run a few trial seeds.

Looking forward to your reply soon. Thanks!
